server/cluster/cluster.go

```diff
@@ -665,6 +664,56 @@ func (c *RaftCluster) processRegionHeartbeat(region *core.RegionInfo) error {
 		time.Sleep(500 * time.Millisecond)
 	})
 
+	return c.saveRegion(isNew, saveKV, saveCache, needSync, region, origin)
+}
+
+// processRegionHeartbeatResponse updates the region information thought region split.
+// It will update peer that not exist in the cache, and add new peers that join in.
+func (c *RaftCluster) processRegionSpiltReport(regions []*metapb.Region) (err error) {
+	if err = checkSplitRegions(regions); err != nil {
+		return err
+	}
+	c.RLock()
+	coreCluster := c.core
+	c.RUnlock()
+	total := len(regions) - 1
+	for i := range regions {
+		// It should update rightmost because it will override by left.
+		index := total - i
+		v := regions[index]
+		if v.GetPeers() == nil {
+			err = errors.Wrap(err, fmt.Sprintf("region:%d peer is nil", v.GetId()))
+			continue
+		}
+		region := core.NewRegionInfo(v, v.Peers[0])
+		origin, perr := coreCluster.PreCheckPutRegion(region)
+		if perr != nil {
+			err = errors.Wrap(err, perr.Error())
+			continue
+		}
+		update, isNew, needSync := false, false, true
+		// The region that is not the rightmost element should update if origin is nil.
+		// The rightmost region should update if epoch greater than origin.
+		if origin == nil {
+			update = true
+		} else if index == total {
+			isNew = false
```

Comment:
When `isNew` will be true?

Reply:
left's origin is exist or rightmost's conf greater than  origin conf .

Reply:
I don't find anywhere to set it to true.

Reply:
yes , it's bug.